Modules/posixmodule.c

```diff
@@ -12544,6 +12544,21 @@ static PyObject *
 os_mkfifo_impl(PyObject *module, path_t *path, int mode, int dir_fd)
 /*[clinic end generated code: output=ce41cfad0e68c940 input=73032e98a36e0e19]*/
 {
+#ifdef MS_WINDOWS
+    HANDLE h;
+    int fd;
+    Py_BEGIN_ALLOW_THREADS
+    h = CreateNamedPipeW(path->wide,
+        PIPE_ACCESS_DUPLEX | FILE_FLAG_FIRST_PIPE_INSTANCE,
+        PIPE_TYPE_BYTE | PIPE_READMODE_BYTE | PIPE_WAIT,
+        1, 1024, 1024, 0, NULL
+    );
+    Py_END_ALLOW_THREADS
+    if (INVALID_HANDLE_VALUE == h)
+        return win32_error_object("CreateNamedPipeW", path->object);
+    fd = _Py_open_osfhandle(h, _O_RDWR);
```

Comment:
`_Py_open_osfhandle()` can return -1: see the Windows implementation of `os.pipe()`.